internet-programming/server/Server.java

```diff
@@ -0,0 +1,60 @@
+package server;
+import java.io.*;
+import java.net.*;
+import java.util.concurrent.*;
+
+public class Server {
+    private static volatile boolean isRunning = true; 
+
+    public static void main(String[] args) throws IOException {
+        ExecutorService pool = Executors.newCachedThreadPool();
+        try (ServerSocket serverSocket = new ServerSocket(31337)) {
+            System.out.println("Server started on port 31337");
+            System.out.println("Type 'EXIT' to shut down the server safely.");
+
+            Thread shutdownThread = new Thread(() -> {
+                BufferedReader consoleReader = new BufferedReader(new InputStreamReader(System.in));
+                try {
+                    while (isRunning) {
+                        String command = consoleReader.readLine();
+                        if (command.equalsIgnoreCase("EXIT")) {
```

Comment:
using "EXIT".equalsIgnoreCase(command) may save you from some NullPointerExceptions